app/models/package_manager/maven.rb

```diff
@@ -210,11 +206,24 @@ def self.latest_version(versions, name)
       end
     end
 
+    def self.parse_names(page)
+      # parse the names from maven-repository.com/artifact/latest pages
```

Comment:
Is it worth having a test with some good and bad names to keep this from breaking?

Reply:
Probably. I'll need to look into how to construct the right object for the page parsing. I'll add a spec for the MavenUrl class in there too to test the `legal_name?` method I added.